src/main/java/com/mraof/minestuck/data/worldgen/MSBiomeProvider.java

```diff
@@ -36,6 +48,9 @@ private static Biome skaiaBiome()
 		BiomeSpecialEffects.Builder ambience = new BiomeSpecialEffects.Builder().waterColor(0x3F76E4).waterFogColor(0x050533);
 		ambience.fogColor(0xCDCDFF).skyColor(0x7AA4FF);
 		
+		genSettings.addFeature(GenerationStep.Decoration.SURFACE_STRUCTURES, Holder.direct(new PlacedFeature(Holder.direct(new ConfiguredFeature<>(MSFeatures.ROOK_OUTPOST.get(), FeatureConfiguration.NONE)),
+				List.of(RarityFilter.onAverageOnceEvery(120), PlacementUtils.HEIGHTMAP_OCEAN_FLOOR, RandomOffsetPlacement.vertical(ConstantInt.of(-4))))));
+		
```

Comment:
I request making use of `MSPlacedFeatureProvider` instead of defining the placed feature here directly. (You can then get your placed-feature-holder by first getting a `HolderGetter<PlacedFeature>` from `context.lookup(Registries.PLACED_FEATURE)`, and then you can get a holder for your resource key from there.)

Reply:
I think there is still a gap in my understanding with this, not sure how to get something from MSPlacedFeatureProvider that I can use in MSBiomeProvider

Reply:
The common link is that both should use the same `ResourceKey<PlacedFeature>` (the ones we already have can be found in `MSPlacedFeatures`). Going from a  `ResourceKey` to a `Holder`, you would use the holder getter I mentioned earlier.

Reply:
I am sorry to say that something is still not clicking, I feel I need to be walked through

Reply:
Should now be in an acceptable form, let me know if you want any further changes!